packages/organizations/origin-organization-irec-api/src/irec/irec.service.ts

```diff
@@ -104,4 +115,203 @@ export class IrecService {
         const irecClient = await this.getIrecClient(user);
         return await irecClient.beneficiary.update(beneficiaryId, params);
     }
+
+    async createDevice(user: UserIdentifier, deviceData: DeviceCreateParams): Promise<IrecDevice> {
+        if (!this.isIrecIntegrationEnabled()) {
```

Comment:
Maybe this is beyond this scope, but IMHO this service is implemented in an ugly way.

If each method is replaced by the "isIrecIntegrationEnabled", then perhaps you should have separate service for that that would be **injected** instead of this primary one IF irec integration is disabled. 

it would make things a lot, a lot cleaner. To make things easier, IrecServiceDisabled could implement public methods of this IrecService, so they are always in sync.

Reply:
done, thanks for the good idea